scss/_functions.scss

```diff
@@ -100,3 +100,44 @@
 
   @return mix($color-base, $color, $level * $theme-color-interval);
 }
+
+// Return valid calc
+@function add($value1, $value2, $return-calc: true) {
+  @if $value1 == null and $value2 == null {
+    @return null;
+  }
+
+  @if $value1 == null {
+    @return $value2;
+  }
+
+  @if $value2 == null {
+    @return $value1;
+  }
+
+  @if type-of($value1) == number and type-of($value2) == number and comparable($value1, $value2) {
```

Comment:
I guess only `comparable()` is good enough here

Reply:
The `type-of($value1) == number and type-of($value2) == number` is necessary to pass nested cases such as:

```scss
.el {
  padding: add($rem, add($rem, $px));
  padding: add($rem, add($rem, $px, false));
}
```

If the condition is missing, a compile error will occur:

```
argument `$number-2` of `comparable($number-1, $number-2)` must be a number ...
```